addons/point_of_sale/models/pos_order.py

```diff
@@ -278,6 +283,49 @@ def _compute_currency_rate(self):
         for order in self:
             order.currency_rate = self.env['res.currency']._get_conversion_rate(order.company_id.currency_id, order.currency_id, order.company_id, order.date_order)
 
+    @api.depends('lines.is_total_cost_computed')
+    def _compute_is_total_cost_computed(self):
+        for order in self:
+            order.is_total_cost_computed = not False in order.lines.mapped('is_total_cost_computed')
+
+    def _compute_total_cost_in_real_time(self):
+        """
+        Compute the total cost of the order when it's processed by the server. It will compute the total cost of all the lines
+        if it's possible. If a margin of one of the order's lines cannot be computed (because of session_id.update_stock_at_closing),
+        then the margin of said order is not computed (it will be computed when closing the session).
+        """
+        for order in self:
+            lines = order.lines
+            anglo_saxon_lines = lines.filtered(lambda l: l.product_id.type == 'product' and l.product_id.cost_method in ['fifo', 'average'])
+            lines -= anglo_saxon_lines
+            compute_anglo_later = bool(anglo_saxon_lines and order.session_id.update_stock_at_closing)
+
+            lines._compute_total_cost()
+            if not compute_anglo_later:
+                stock_moves = order.picking_ids and order.picking_ids.move_lines
+                anglo_saxon_lines._compute_total_cost(stock_moves)
+
+    def _compute_total_cost_at_session_closing(self, stock_moves):
+        """
+        Compute the margin at the end of the session. This method should be called to compute the remaining lines margin
+        which are anglo saxon and then compute the order margin
+        :param stock_moves: record set of `stock.move` from the session
+        """
+        for order in self:
+            anglo_saxon_lines = order.lines.filtered(lambda l: l.product_id.type == 'product' and l.product_id.cost_method in ['fifo', 'average'])
```

Comment:
anglo_saxon and cost method are two different things, so maybe don't name it like that

Reply:
Changed